### PR TITLE
[2481] Column mapping removal check for invalid column names

### PR DESCRIPTION
**Pull request by sabir-akhadov**
Add new column mapping removal command basic class hidden behind a feature flag.
Add a check for invalid column names before applying the command.
#### Which Delta project/connector is this regarding?

- [x] Spark
- [ ] Standalone
- [ ] Flink
- [ ] Kernel
- [ ] Other (fill in here)

## How was this patch tested?
new tests

## Does this PR introduce _any_ user-facing changes?

No

### CODE REVIEW

File: spark/src/main/resources/error/delta-error-classes.json [hunk 1]
```diff
@@ -1130,6 +1130,15 @@
     ],
     "sqlState" : "42K05"
   },
+  "DELTA_INVALID_CHARACTERS_IN_COLUMN_NAMES_UPON_SETTING_COLUMN_MAPPING_TO_NONE" : {
```
Comment (tomvanbussel) [inline spark/src/main/resources/error/delta-error-classes.json:1133]: Nit: Let's maybe use a slightly error class.

**Review by tomvanbussel — APPROVED** [review]
One small nit, LGTM otherwise

File: spark/src/main/resources/error/delta-error-classes.json (hunk 1 above)
Comment (sabir-akhadov) [reply spark/src/main/resources/error/delta-error-classes.json:1133]: Is `DELTA_INVALID_COLUMN_NAMES_WHEN_REMOVING_COLUMN_MAPPING` better?

Comment (tomvanbussel) [reply spark/src/main/resources/error/delta-error-classes.json:1133]: SGTM

**Pull request closed without merging**